File: ConexionBD/UsuarioBD.cs
using punto_de_venta_C_.Modelo;
using MySqlConnector;
using System;
using System.Collections.Generic;

namespace punto_de_venta_C_.ConexionBD
{
    public class UsuarioBD
    {
        public List<Usuario> obtenerTodos()
        {
            List<Usuario> lista = new List<Usuario>();
            string sql = "SELECT * FROM usuarios ORDER BY id";
            try
            {
                MySqlConnection c = Conexion.getConexion();
                using (MySqlCommand st = new MySqlCommand(sql, c))
                {
                    using (MySqlDataReader rs = st.ExecuteReader())
                    {
                        while (rs.Read())
                        {
                            lista.Add(mapear(rs));
                        }
                    }
                }
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
            return lista;
        }

        public Usuario autenticar(string username, string password)
        {
            string sql = "SELECT * FROM usuarios WHERE username = @username AND password = @password";
            try
            {
                MySqlConnection c = Conexion.getConexion();
                using (MySqlCommand ps = new MySqlCommand(sql, c))
                {
                    ps.Parameters.AddWithValue("@username", username);
                    ps.Parameters.AddWithValue("@password", password);
                    using (MySqlDataReader rs = ps.ExecuteReader())
                    {
                        if (rs.Read())
                            return mapear(rs);
                    }
                }
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
            return null;
        }

        public bool insertar(Usuario u)
        {
            string sql = "INSERT INTO usuarios (username, password, rol, nombre, edad, sexo) VALUES (@username,@password,@rol,@nombre,@edad,@sexo)";
            try
            {
                MySqlConnection c = Conexion.getConexion();
                using (MySqlCommand ps = new MySqlCommand(sql, c))
                {
                    ps.Parameters.AddWithValue("@username", u.getUsername());
                    ps.Parameters.AddWithValue("@password", u.getPassword());
                    ps.Parameters.AddWithValue("@rol", u.getRol());
                    ps.Parameters.AddWithValue("@nombre", u.getNombre());
                    ps.Parameters.AddWithValue("@edad", u.getEdad());
                    ps.Parameters.AddWithValue("@sexo", u.getSexo());
                    int rows = ps.ExecuteNonQuery();
                    if (rows > 0)
                    {
                        u.setId((int)ps.LastInsertedId);
                        return true;
                    }
                }
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
            return false;
        }

        public bool actualizar(Usuario u)
        {
            string sql = "UPDATE usuarios SET username=@username, password=@password, rol=@rol, nombre=@nombre, edad=@edad, sexo=@sexo WHERE id=@id";
            try
            {
                MySqlConnection c = Conexion.getConexion();
                using (MySqlCommand ps = new MySqlCommand(sql, c))
                {
                    ps.Parameters.AddWithValue("@username", u.getUsername());
                    ps.Parameters.AddWithValue("@password", u.getPassword());
                    ps.Parameters.AddWithValue("@rol", u.getRol());
                    ps.Parameters.AddWithValue("@nombre", u.getNombre());
                    ps.Parameters.AddWithValue("@edad", u.getEdad());
                    ps.Parameters.AddWithValue("@sexo", u.getSexo());
                    ps.Parameters.AddWithValue("@id", u.getId());
                    return ps.ExecuteNonQuery() > 0;
                }
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
            return false;
        }

        public bool eliminar(int id)
        {
            string sql = "DELETE FROM usuarios WHERE id = @id";
            try
            {
                MySqlConnection c = Conexion.getConexion();
                using (MySqlCommand ps = new MySqlCommand(sql, c))
                {
                    ps.Parameters.AddWithValue("@id", id);
                    return ps.ExecuteNonQuery() > 0;
                }
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
            return false;
        }

        private Usuario mapear(MySqlDataReader rs)
        {
            return new Usuario(rs.GetInt32("id"), rs.GetString("username"), rs.GetString("password"), rs.GetString("rol"),
                rs.GetString("nombre"), rs.GetInt32("edad"), rs.GetString("sexo"));
        }
    }
}

File: ConexionBD/UsuarioBD.py
from ConexionBD.Conexion import Conexion
from Modelo.Usuario import Usuario

class UsuarioBD:
    def obtenerTodos(self):
        lista = []
        sql = "SELECT * FROM usuarios ORDER BY id"
        try:
            conn = Conexion.getConexion()
            if conn:
                cursor = conn.cursor(dictionary=True)
                cursor.execute(sql)
                rows = cursor.fetchall()
                cursor.close()
                for row in rows:
                    lista.append(self.mapear(row))
        except Exception as e:
            print(e)
        return lista

    def autenticar(self, username, password):
        sql = "SELECT * FROM usuarios WHERE username = %s AND password = %s"
        try:
            conn = Conexion.getConexion()
            if conn:
                cursor = conn.cursor(dictionary=True)
                cursor.execute(sql, (username, password))
                row = cursor.fetchone()
                cursor.close()
                if row:
                    return self.mapear(row)
        except Exception as e:
            print(e)
        return None

    def insertar(self, u):
        sql = "INSERT INTO usuarios (username, password, rol, nombre, edad, sexo) VALUES (%s,%s,%s,%s,%s,%s)"
        try:
            conn = Conexion.getConexion()
            if conn:
                cursor = conn.cursor()
                cursor.execute(sql, (u.getUsername(), u.getPassword(), u.getRol(), u.getNombre(), u.getEdad(), u.getSexo()))
                conn.commit()
                if cursor.rowcount > 0:
                    u.setId(cursor.lastrowid)
                    cursor.close()
                    return True
                cursor.close()
        except Exception as e:
            print(e)
        return False

    def actualizar(self, u):
        sql = "UPDATE usuarios SET username=%s, password=%s, rol=%s, nombre=%s, edad=%s, sexo=%s WHERE id=%s"
        try:
            conn = Conexion.getConexion()
            if conn:
                cursor = conn.cursor()
                cursor.execute(sql, (u.getUsername(), u.getPassword(), u.getRol(), u.getNombre(), u.getEdad(), u.getSexo(), u.getId()))
                conn.commit()
                res = cursor.rowcount > 0
                cursor.close()
                return res
        except Exception as e:
            print(e)
        return False

    def eliminar(self, id_usuario):
        sql = "DELETE FROM usuarios WHERE id = %s"
        try:
            conn = Conexion.getConexion()
            if conn:
                cursor = conn.cursor()
                cursor.execute(sql, (id_usuario,))
                conn.commit()
                res = cursor.rowcount > 0
                cursor.close()
                return res
        except Exception as e:
            print(e)
        return False

    def mapear(self, row):
        u = Usuario()
        u.setId(row["id"])
        u.setUsername(row["username"])
        u.setPassword(row["password"])
        u.setRol(row["rol"])
        u.setNombre(row["nombre"])
        u.setEdad(row["edad"])
        u.setSexo(row["sexo"])
        return u
